addons/stock/models/product_strategy.py

```diff
@@ -12,47 +12,36 @@ class RemovalStrategy(models.Model):
     method = fields.Char("Method", required=True, help="FIFO, LIFO...")
 
 
-class PutAwayStrategy(models.Model):
+class PutAwayRule(models.Model):
     _name = 'product.putaway'
-    _description = 'Put Away Strategy'
-
-    name = fields.Char('Name', required=True)
-
-    fixed_location_ids = fields.One2many(
-        'stock.fixed.putaway.strat', 'putaway_id',
-        'Fixed Locations Per Product Category', domain=[('category_id', '!=', False)], copy=True)
-    product_location_ids = fields.One2many(
-        'stock.fixed.putaway.strat', 'putaway_id',
-        'Fixed Locations Per Product', domain=[('product_id', '!=', False)], copy=True)
-
-    def putaway_apply(self, product):
-        put_away = self._get_putaway_rule(product)
-        if put_away:
-            return put_away.fixed_location_id
-        return self.env['stock.location']
-
-    def _get_putaway_rule(self, product):
-        if self.product_location_ids:
-            put_away = self.product_location_ids.filtered(lambda x: x.product_id == product)
-            if put_away:
-                return put_away[0]
-        if self.fixed_location_ids:
-            categ = product.categ_id
-            while categ:
-                put_away = self.fixed_location_ids.filtered(lambda x: x.category_id == categ)
-                if put_away:
-                    return put_away[0]
-                categ = categ.parent_id
-        return self.env['stock.location']
-
-
-class FixedPutAwayStrategy(models.Model):
-    _name = 'stock.fixed.putaway.strat'
     _order = 'sequence'
```

Comment:
you'll have a listview with putaway rule for product en categories mixed, it would be nice to always dispay the rule on product and then the rules en category even if they have the same sequence
not sure how we do null first but 'sequence, product_id' could do the trick